examples/tv-casting-app/APIs.md

```diff
@@ -0,0 +1,692 @@
+            return APP_ID.getBytes();
+        }
+    }
+    ```
+
+    On iOS, define the
+    `func castingAppDidReceiveRequestForRotatingDeviceIdUniqueId` in a class,
+    `AppParametersDataSource`, that implements the `MTRDataSource`:
+
+    ```objectivec
+    class AppParametersDataSource : NSObject, MTRDataSource
+    {
+        func castingAppDidReceiveRequestForRotatingDeviceIdUniqueId(_ sender: Any) -> Data {
+            return "YOUR_GENERATED_ROTATING_DEVICE_ID".data(using: .utf8)!      // dummy value for demonstration only
+        }
+        ...
+    }
+    ```
+
+2.  **Commissioning Data** - This object contains the passcode, discriminator,
+    etc which identify the app and are provided to the CastingPlayer during the
+    commissioning process. Refer to the Matter specification's
+    [Onboarding Payload](https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/qr_code/OnboardingPayload.adoc#ref_OnboardingPayload))
+    section for details on commissioning data.
+
+    On Android, define a `commissioningDataProvider` that can provide the
+    required values to the CastingApp.
+
+    ```java
+    private final static DataProvider<CommissioningData> commissioningDataProvider = () -> {
+        CommissioningData commissioningData = new CommissioningData();
+        commissioningData.setSetupPasscode(20202021);   // dummy values for demonstration only
+        commissioningData.setDiscriminator(3874);
+        return commissioningData;
+    };
+    ```
+
+    On iOS, add a `func commissioningDataProvider` to the
+    `AppParametersDataSource` class defined above, that can provide the required
+    values to the `MTRCastingApp`.
+
+    ```objectivec
+    func castingAppDidReceiveRequestForCommissioningData(_ sender: Any) -> MTRCommissioningData {
+        return MTRCommissioningData(
+            passcode: 20202021,         // dummy values for demonstration only
+            discriminator: 3874,
+            spake2pIterationCount: 1000,
+            spake2pVerifier: nil,
+            spake2pSalt: nil)
+    }
+    ```
+
+3.  **Device Attestation Credentials** - This object contains the
+    `DeviceAttestationCertificate`, `ProductAttestationIntermediateCertificate`,
+    etc. and implements a way to sign messages when called upon by the Matter TV
+    Casting Library as part of the
+    [Device Attestation process](https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/device_attestation/Device_Attestation_Specification.adoc)
+    during commissoning.
+
+    On Android, define a `dacProvider` to provide the Casting Client's Device
+    Attestation Credentials, by implementing a `DataSource`:
+
+    ```java
+    private final static DataProvider<DeviceAttestationCredentials> dacProvider = new DataProvider<DeviceAttestationCredentials>() {
+        private static final String kDevelopmentDAC_Cert_FFF1_8001 = "MIIB5z...<snipped>...CXE1M=";     // dummy values for demonstration only
```

Comment:
if all 3 are dummy values for demonstration only, then put the comment above the first one. Otherwise, it could be interpreted to mean that only this first variable is dummy value.

Reply:
Updated in project-chip#29444